pkg/cluster/fetch_test.go

```diff
@@ -32,3 +32,85 @@ func TestGetBundlesForCluster(t *testing.T) {
 	g.Expect(err).To(BeNil())
 	g.Expect(gotBundles).To(Equal(wantBundles))
 }
+
+func TestGetFluxConfigForClusterIsNil(t *testing.T) {
+	g := NewWithT(t)
+	c := &v1alpha1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "eksa-cluster",
+			Namespace: "eksa",
+		},
+		Spec: v1alpha1.ClusterSpec{
+			GitOpsRef: nil,
+		},
+	}
+	var wantFlux *v1alpha1.FluxConfig
+	mockFetch := func(ctx context.Context, name, namespace string) (*v1alpha1.FluxConfig, error) {
+		g.Expect(name).To(Equal(c.Name))
+		g.Expect(namespace).To(Equal(c.Namespace))
+
+		return wantFlux, nil
+	}
+
+	gotFlux, err := cluster.GetFluxConfigForCluster(context.Background(), c, mockFetch)
+	g.Expect(err).To(BeNil())
+	g.Expect(gotFlux).To(Equal(wantFlux))
+}
+
+func TestGetFluxConfigForCluster(t *testing.T) {
+	g := NewWithT(t)
+	c := &v1alpha1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "eksa-cluster",
+			Namespace: "eksa",
+		},
+		Spec: v1alpha1.ClusterSpec{
+			GitOpsRef: &v1alpha1.Ref{
+				Kind: v1alpha1.FluxConfigKind,
+				Name: "eksa-cluster",
+			},
+		},
+	}
+	wantFlux := &v1alpha1.FluxConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "",
+			APIVersion: "",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:                       "",
```

Comment:
We don't list to list these fields then right? It would be cleaner to just specify an empty struct.

Reply:
fixed